vslam/feature/orb_feature.py

```diff
@@ -0,0 +1,205 @@
+        
+        return total_kps
+            
+    def _keypointFilter(self, fastKps: list, imgSize : Tuple, numKps : int) -> list:
+        """Filter Keypoints
+            Args: 
+                Keypoints
+                Maximum Number of Keypoint
+
+            Returns:
+                Reduced Keypoints
+        """
+        edge_removed = self._runByImageBorder(fastKps, imgSize)
+        retained_best = self._retainBest(edge_removed, int(numKps))
+        return retained_best
+
+    def _runByImageBorder(self, kps: list, imgSize : Tuple) -> List:
+        """Remove keypoints too close to the image border
+            Args:
+                Keypoints
+                Image Size
+                EdgeThreshold
+
+            Return:
+                Keypoints not immediatley close to image border
+        """
+
+        # TODO: This might be slow
+        (h, w) = imgSize
+        inner_border = [k for k in kps if ((k.pt[0] > self.edgeThreshold) and (k.pt[0] < w - self.edgeThreshold)) and \
+             ((k.pt[1] > self.edgeThreshold) and (k.pt[1] < h - self.edgeThreshold))]
+        return inner_border
+
+    def _retainBest(self, kps : list, npoints : int) -> list:
+        """Keypoint filter to retain best kps
+            Args:
+                Keypoints
+                Limited Points
+
+            Return:
+                Keypoints
+        """
+        if len(kps) < npoints:
+            return kps
+        
+        if npoints == 0:
+            return kps
+
+        kps.sort(key= lambda x : x.response, reverse=True)
+        return kps[:npoints]        
+    
+    def _computePyramid(self, img : np.ndarray) -> List:
+        """Compute Pyramid for incoming images
+            Args:
+                Grey scale image
+            
+            Returns:
+                list of Gaussian Pyramid
+                Level 0 is original Resolution
+        """
+        scale = 1
+        (h, w) = img.shape
+        imgPyrd = list()
+        for i in range(self.nlevels):
+            invScale = 1. / scale
+            temp = cv2.resize(img, (np.round(w * invScale).astype(int), np.round(h * invScale).astype(int)), interpolation=cv2.INTER_LINEAR)
+            # This function auto pads around
+            wholeSize = cv2.copyMakeBorder(temp, self.edgeThreshold, self.edgeThreshold,
+                self.edgeThreshold, self.edgeThreshold,
+                cv2.BORDER_REFLECT_101+cv2.BORDER_ISOLATED)
+            imgPyrd.append(wholeSize)
+            scale *= self.scaleFactor
+        return imgPyrd
+
+    def detectAndCompute(self, img: np.ndarray, mask : np.ndarray = None) -> Tuple[list, np.ndarray]:
+        """Wrapper class for ORB detectAndCompute
+            Args:
+                Image
+                Mask
+
+            Returns:
+                Tuple of Keypoint and descriptors
+        """
+        img = cv2.cvtColor(img, cv2.COLOR_BGR2GRAY)
+        imgPyrd = self._computePyramid(img)
```

Comment:
I wonder why this necessary? Doesn't orb detectAndCompute already detect features at different scales?

I think this code is still useful for when we write ORB2FeatureExtractor. What do you think?